velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -1412,9 +1412,18 @@ PlanBuilder& PlanBuilder::window(
 
 PlanBuilder& PlanBuilder::rowNumber(
     const std::vector<std::string>& partitionKeys,
-    std::optional<int32_t> limit) {
+    std::optional<int32_t> limit,
+    const bool generateRowNumber) {
```

Comment:
Any reason to not use std::optional for rowNumberColumnName ? That's how the TopNRowNumberNode implements also. And the column name wouldn't be hard-coded.

Reply:
This is the test PlanBuilder for RowNumber node.
The corresponding signature for TopNRow is
```
PlanBuilder& PlanBuilder::topNRowNumber(
    const std::vector<std::string>& partitionKeys,
    const std::vector<std::string>& sortingKeys,
    int32_t limit,
    bool generateRowNumber) {
```
It is the same approach using a boolean to indicate whether or not the node constructor gets passed the column name via the optional.
The actual RowNumber node constructor does use an optional just like the TopNRowNumber node.

Am I missing something?

Reply:
My bad... I misread this. For PlanBuilder that is fine.